development/wlopm/wlopm.info

```diff
@@ -0,0 +1,10 @@
+PRGNAM="wlopm"
+VERSION="0.1.0"
+HOMEPAGE="https://git.sr.ht/~leon_plickat/wlopm/"
+DOWNLOAD="https://git.sr.ht/~leon_plickat/wlopm/archive/v0.1.0.tar.gz"
```

Comment:
The links from git.sr.ht are tricky since I don't think there is a way to make it always consistently give the same filename with a browser and with wget or with wget respecting content disposition.

Can you handle both file names in your `tar xvf` step?

```
curl -o /dev/null -v https://git.sr.ht/~leon_plickat/wlopm/archive/v0.1.0.tar.gz 2>&1  | grep -i content-disposition
< content-disposition: attachment; filename=wlopm-v0.1.0.tar.gz
```

so either wlopm-v0.1.0.tar.gz or v0.1.0.tar.gz?

Reply:
Looking at previously published slackbuilds that are dependent off sources on git.sr.ht, would the following then suffice?

```
diff --git a/desktop/wlopm/wlopm.SlackBuild b/desktop/wlopm/wlopm.SlackBuild
index bb7c6e3..4ba8498 100644
--- a/desktop/wlopm/wlopm.SlackBuild
+++ b/desktop/wlopm/wlopm.SlackBuild
@@ -67,7 +67,7 @@ rm -rf $PKG
 mkdir -p $TMP $PKG $OUTPUT
 cd $TMP
 rm -rf $PRGNAM-v$VERSION
-tar xvf $CWD/v$VERSION.tar.gz
+tar xvf $CWD/$PRGNAM-v$VERSION.tar.gz || tar xvf $CWD/v$VERSION.tar.gz
 cd $PRGNAM-v$VERSION
 chown -R root:root .
 find -L . \
```

```
$ for i in `find . -name *.info | xargs grep -E "DOWNLOAD=.*git.sr.ht" | cut -f3 -d'/'`; do find . -name $i.SlackBuild | xargs grep -RE "tar "; done
tar xvf $CWD/${PRGNAM}-${VERSION}.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$VERSION.tar.gz || tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
    tar xvf $CWD/$dep-$ver.crate
tar xvf $CWD/$PRGNAM-v$VERSION.tar.gz || tar xvf $CWD/v$VERSION.tar.gz # not possible to get consistent filenames with sourcehut
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || $CWD/$VERSION.tar.gz
tar xvf $CWD/$VERSION.tar.gz || tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
    tar xvf $CWD/$dep-$ver.crate
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$PRGNAM-v$VERSION.tar.gz || tar xvf $CWD/v$VERSION.tar.gz
tar xvf $CWD/$VERSION.tar.?z || tar xvf $CWD/$PRGNAM-$VERSION.tar.?z
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$SRCNAM.tar.gz || tar xvf $CWD/v$VERSION.tar.gz
tar xvf $CWD/$SRCNAM.tar.gz || tar xvf $CWD/v$VERSION.tar.gz
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/$VERSION.tar.gz
tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
tar xvf $CWD/$SRCNAM-$VERSION.tar.gz
tar xvf $CWD/$VERSION.tar.gz || tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
```